.github/workflows/ci.yml

```diff
@@ -73,6 +73,8 @@ jobs:
           platforms: arm64
       - name: Test Docker Image
         run: core/docker/build.sh
+      - name: Block Merge Commits
```

Comment:
`Block Merge/Fixup Commits` - that's what your action seems to do (and also `!squash` commits but I've not seen people use that as often).

Reply:
Also should this be separate step (relatively quick) and will not confuse users that compilation failed?

Reply:
Do you mean a separate job? I think so, I was considering this.

Reply:
Yes, meant a job. Thanks for catching.

Reply:
I'll also try adding a review with a change request instead of failing the CI to differentiate this a bit better. If that works, it can stay as a step here, but I'll move it to the top.